src/routing/approved.rs

```diff
@@ -665,7 +668,38 @@ impl Approved {
                 self.handle_join_request(msg.src().to_node_peer(sender)?, *join_request.clone())
             }
             Variant::UserMessage(content) => {
-                self.handle_user_message(msg.src().src_location(), *msg.dst(), content.clone())
+                if msg.dst() == &DstLocation::AccumulatingNode(self.node().name()) {
```

Comment:
I'd put this new logic into `handle_user_message`, to keep this function small (-ish).